crates/biome_cli/src/execute/traverse.rs

```diff
@@ -309,6 +309,8 @@ fn process_messages(options: ProcessMessagesOptions) {
 
     let mut paths: FxHashSet<String> = FxHashSet::default();
     let mut printed_diagnostics: u16 = 0;
+    let mut printed_error_diagnostics: u16 = 0;
+    let mut printed_warning_diagnostics: u16 = 0;
```

Comment:
I see an issue here. Diagnostics aren't just warnings or errors. Diagnostics have **many** levels of severity:

https://github.com/biomejs/biome/blob/fb2bce9c4a14dd5c30a2dbae2ac5b01b6604b6db/crates/biome_diagnostics/src/diagnostic.rs#L121-L133

I think we should not keep count of warnings and errors.

I think the solution is simpler:

```rs
let shout_print = printed_diagnostics < max_diagnostics && diag.severity() >= *diagnostic_level;
if should_print {
	printed_diagnostics += 1
}
```

Essentially, before increasing `printed_diagnostics`, we should check if the diagnosing we received from the thread has a severity equals or higher than `diagnostic_level`, which is the level set by the user.

Reply:
@yossydev yes, but you have to repeat this logic in all the places where you have `printed_diagnostics += 1`

Reply:
@ematipico 
Thank you! I have added an additional fix with https://github.com/biomejs/biome/pull/1248/commits/195639ce668b8adfe4772af8048ded3c3818c6f5!

I just haven't been able to fix the other two...🙏🏻 

https://github.com/biomejs/biome/blob/28f2de482625e2ce650b407637981720d1298df9/crates/biome_cli/src/execute/traverse.rs#L347-L349

https://github.com/biomejs/biome/blob/28f2de482625e2ce650b407637981720d1298df9/crates/biome_cli/src/execute/traverse.rs#L505

Do I need to edit the `AppryError` and `Diff` structures to use `.severity()`?👀

Reply:
Yes @yossydev  :)

**All** places where we increase the printed diagnostics

Reply:
Thanks for your comment!

I did the rest of the modifications at https://github.com/biomejs/biome/pull/1248/commits/0869b03d92b298cd8b0ad0cefa120aeaf731d8c6. Is this the right place to do it?👀

Reply:
@ematipico 
Thanks for your comment!

I did the rest of the modifications at https://github.com/biomejs/biome/pull/1248/commits/0869b03d92b298cd8b0ad0cefa120aeaf731d8c6. Is this the right place to do it?👀